### PR TITLE
Support Squash merges

### PR DESCRIPTION
**Pull request by tmspicer**
### SUMMARY

The following will support determining the PR number from a squash merge.  It currently depends on the default merge commit message syntax for a squash merge.

To be able to support none standard commit messages, some more logic would need to go in to use the API to query the PR by the commit SHA.

Addresses https://github.com/jefflinse/pr-semver-bump/issues/15

### RELEASE NOTES

* Support Squash merges for default squash merge message

### CODE REVIEW

File: pr.test.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ const { extractPRNumber, getReleaseType, getReleaseNotes, fetchPR } = require('.
 test('can extract a PR number from a PR merge commit message', () => {
     expect(extractPRNumber('Merge pull request #4 from some/mockBranch')).toEqual('4')
     expect(extractPRNumber('Merge pull request #42 from some/mockBranch')).toEqual('42')
+    expect(extractPRNumber('My PR squashed (#56)')).toEqual('56')
+    expect(extractPRNumber('Squash pr (#45)\n\n multiple commit messages appended')).toEqual('45')
```
Comment (jefflinse) [inline pr.test.js:7]: Since this unit test is specific to merge commits, let's move these two cases into their own new test for squash merges.

**Review by jefflinse — CHANGES_REQUESTED** [review]
Hi @tmspicer, thanks for opening this issue (and PR)! I'm happy to take this change, just one organizational nitpic and I think it's good to go. Dev CI won't run for fork PRs, so I verified your changes locally.

File: pr.test.js (hunk 1 above)
Comment (tmspicer) [reply pr.test.js:10]: I've split the pr message testcase out for the squash merge case.

**Comment by jefflinse** [comment]
LGTM

**Review by jefflinse — APPROVED** [review]

**Pull request merged**